### ISSUE TITLE
Syntax [![](image_location)](link_target) for making image links?

### ISSUE THREAD

**Issue by np-8** [phseiff/github-flavored-markdown-to-html#26]
This is quite simple thing. There are [few ways](https://meta.stackexchange.com/questions/38915/creating-an-image-link-in-markdown-format) to make images work as links. Two of the following syntaxes do work (`gh-md-to-html` v. 1.7.0), but the last one, which is in my opinion the prettiest, as it is the least verbose, does not. 

Is there a reason this syntax does not work, and could that syntax be supported in the future? I'm using the `OFFLINE+` core_converter.

### Test markdown code

```
Works 

<a href="images/test.png">![](images/test.png)</a>

Also works

[<img src="images/test.png">](images/test.png)

Does not work

[![](images/test.png)](images/test.png)
```

**Comment by phseiff** [comment]
Thanks for pointing this out to me!

I tested whether these Syntaxes work on GitHub below, and it appears GitHub's behavior, in this regard, is identical to `gh-md-to-html`'s behavior:

Works (`<a href="images/test.png">![](https://avatars.githubusercontent.com/u/31518703?v=4)</a>`):
<a href="images/test.png">![](https://avatars.githubusercontent.com/u/31518703?v=4)</a>

Also works (`[<img src="https://avatars.githubusercontent.com/u/31518703?v=4">](https://avatars.githubusercontent.com/u/31518703?v=4)`):
[<img src="https://avatars.githubusercontent.com/u/31518703?v=4">](https://avatars.githubusercontent.com/u/31518703?v=4)

Does not work (`[![](images/test.png)](https://avatars.githubusercontent.com/u/31518703?v=4)`):
[![](images/test.png)](https://avatars.githubusercontent.com/u/31518703?v=4)
(emptyness)

---

> Is there a reason this syntax does not work, and could that syntax be supported in the future?
I wasn't even aware that this doesn't work (it feels a little odd to me, too), but the reason why it doesn't work in OFFLINE-mode is that mistune, apparently, doesn't support it, and the reason that it (probably) doesn't work in non-offline mode, too, is that (as shown above) GitHub's API doesn't support it.

I think that this is probably part of the markdown specification, so I'm not sure if allowing it would even be desirable, and it would probably be difficult to implement since it requires doing some markdown parsing, which is something this module completely avoided so far. Also, users can use non-standard markdown syntax by plugging a different core converter that allows this syntax into gh-md-to-html.

I'll look some more into this at a later time.

**Comment by np-8** [comment]
Here is a simple test with the syntax that works at least in the preview mode:

```
[![](https://homepages.cae.wisc.edu/~ece533/images/watch.png)](http://www.stackoverflow.com)
```

Producing:
[![](https://homepages.cae.wisc.edu/~ece533/images/watch.png)](http://www.stackoverflow.com)


I also tested with 

```markdown
# example.md
[![](images/local_image.png)](images/local_image.png)

[![](https://someaddress/image.png)](images/local_image.png)

[![](images/local_image.png)](http://www.example.com)
```

when using the default `core_converter` (GitHub API) for `gh_md_to_html.main()` and they **_all did work_**. So probably this is just an issue with the mistune or integration with mistune.

I think that the `[![alt](img_source)](link_address)` syntax is nice since one can use only markdown, and does not have to use html tags to add a link or an image.

**Comment by phseiff** [comment]
> Here is a simple test with the syntax that works at least in the preview mode:
> 
> ```
> [![](https://homepages.cae.wisc.edu/~ece533/images/watch.png)](http://www.stackoverflow.com)
> ```
> 
> Producing:
> [image]

Oh okay, I see; it seems I made a typo when I [tested whether it works in preview mode](https://github.com/phseiff/github-flavored-markdown-to-html/issues/26#issuecomment-812913424) before.

> when using the default `core_converter` (GitHub API) for `gh_md_to_html.main()` and they **_all did work_**. So probably this is just an issue with the mistune or integration with mistune.

It seems like it.... I will need to fix this in OFFLINE mode, then (maybe there's a setting in mistune I can tweak, or a mistune plugin fro that, or I will have to work hands-on with the parser).

---

Thanks for pointing this out to me! :D

**Comment by phseiff** [comment]
I just tested this on my locale machine, and I wast able to reproduce this behavior (though I apparently misunderstood your problem description initially, since I assumed both images wouldn't be shown at all).
On my machine,

```
![](https://homepages.cae.wisc.edu/~ece533/images/airplane.png)

[![](https://homepages.cae.wisc.edu/~ece533/images/airplane.png)](https://github.com/phseiff)
```

are rendered to two correct images, both when using `OFFLINE` as well as `OFFLINE+` and when omitting them;
however, using `OFFLINE+` and `OFFLINE` results images that miss their links.

This appears to be a bug in either the `core_converter`, or the post processing of the generated HTML clashes with the core converter's behavior, yet not the online converter's.

**Comment by phseiff** [comment]
@np-8 I fixed this now; you can update to v1.7.4 to have `OFFLINE` and `OFFLINE+` behave just like the online version.
The old behavior was, indeed, a bug.

(In case you wonder why I closed and re-opened your other issue, I mistyped the issue number in the commit message so it closed the wrong issue.)
